Release.bug fixes & Improvements

Huasito-Appel opened the pull request:
Should have splited into more commits.
TLDR 
Add intructions regarding proxy usage for anthropics api calls
Add UI help for fields
Add cost estimations for Api Calls
Improve Promp formatting and handling on the tool
Better concurrency handle
Timeout on promise to avoid hanging
Handle of failed requests

Here are some screenshots of how it looks.

- base
- ![Screenshot 2025-05-26 at 11 17 05 AM](https://github.com/user-attachments/assets/d1d8ce32-c906-4c97-a217-f166a58c4de6)
- ![Screenshot 2025-05-26 at 11 17 13 AM](https://github.com/user-attachments/assets/1ff2acbc-51af-4db9-834a-d38e79d12f30)
- settings menu
- ![Screenshot 2025-05-26 at 11 17 23 AM](https://github.com/user-attachments/assets/5ee44bb9-754c-4c59-bd53-0b3501d84817)
- Retry Pop Up
- ![Screenshot 2025-05-26 at 11 16 25 AM](https://github.com/user-attachments/assets/896b0863-a74b-4665-95ba-42f6cd850c33)

Commit:
feat:
Add intructions regarding proxy usage for anthropics api calls
Add UI help for fields
Add cost estimations for Api Calls
Improve Promp formatting and handling on the tool
Better concurrency handle
Timeout on promise to avoid hanging
Handle of failed requests

Huasito-Appel commented on lib/evaluateApplicants.ts:
```diff
@@ -278,23 +288,42 @@ const evaluateApplicant = async (
+          retries: 3,
+          factor: 2,
+          minTimeout: 1000,
+          maxTimeout: 5000,
```
Currently this is not configurable, we may want to do it on the same settings page.

agucova replied:
Yo creo que tener un buen default acá puede ser suficiente.

Huasito-Appel commented on lib/prompts/promptTemplates.ts:
```diff
@@ -0,0 +1,55 @@
+// Prompt template system for AI evaluations
```
Here are the templates, now they are configurable and people can change it on the settings, let me know if we want to add more of them per default

agucova commented on README.md:
```diff
@@ -26,16 +41,72 @@ To start developing this extension:
+- OpenAI allows cross-origin requests from Airtable extensions
+- **Setup**: Just configure your API key - no additional steps needed
+
+### Anthropic: Proxy Server Required 🔄
```
Creo que podemos obsoletar el proxy completamente simplemente agregando el header `anthropic-dangerous-direct-browser-access: true` a las llamadas a la API de Anthropic. Mira [aqui](https://simonwillison.net/2024/Aug/23/anthropic-dangerous-direct-browser-access/).

agucova commented on frontend/components/SettingsDialog.tsx:
```diff
@@ -135,6 +212,117 @@ export const SettingsDialog = ({ isOpen, onClose }: { isOpen: boolean; onClose:
+                ]}
+                value={preferredCurrency}
+                onChange={value => setPreferredCurrency(value as string)}
+              />
```
Honestamente, siento que esta feature es innecesaria. Todos los usuarios se manejan en dólares.

agucova commented on frontend/components/SettingsDialog.tsx:
```diff
@@ -135,6 +212,117 @@ export const SettingsDialog = ({ isOpen, onClose }: { isOpen: boolean; onClose:
+              />
+              <div className="mt-1 text-xs text-gray-500">
+                Number of simultaneous API calls. Higher values are faster but may hit rate limits.
+              </div>
```
Una pregunta es si realmente necesitamos esto, vs simplemente ser optimista y partir con una rate alta y hacer que los response headers hagan el trabajo de regular el rate dinámicamente. Onda creo que tanto Anthropic como OpenAI permiten dinámicamente leer los rate limits restantes y los requests per minute posibles. Mira [aquí](https://platform.openai.com/docs/guides/rate-limits) y [aquí](https://docs.anthropic.com/en/api/rate-limits#response-headers). Si hay que decidir si basarse en feedback (creo que la implementación actual ya hace esto, en base a Too Many Requests), o basarse en los RPM de los headers.

Por ahora esto está bien.

agucova commented on README.md:
```diff
@@ -12,6 +12,21 @@ In a [previous pilot project](https://github.com/bluedotimpact/ai-application-ev
+3. Start proxy: `bun run proxy.ts` (keep running)
+4. In new terminal: `bun run start:applications`
+5. ✅ Ready to evaluate! (Uses `claude-3-5-haiku` by default)
+
```
No falta pedir configurar el blockId y baseId?

agucova replied:
Tampoco explica como agregar la extension en Airtable

Commit:
lint: ran biome linter & formatter

Commit:
chore: add pre commit hook for linter

Commit:
feat: remove proxy and add direct API calls with rate limit headers

- Remove proxy.ts and proxy-related scripts
- Update Anthropic/OpenAI APIs to use direct calls
- Add rate limit header logging for monitoring
- Remove currency conversion, default to USD only
- Update README with correct BlockId/BaseId instructions
